presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -343,13 +350,32 @@ protected ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaData met
         if (mapping.isPresent()) {
             return mapping;
         }
-        Optional<ColumnMapping> connectorMapping = jdbcTypeToPrestoType(typeHandle);
-        if (connectorMapping.isPresent()) {
-            return connectorMapping;
+
+        mapping = jdbcTypeToPrestoType(typeHandle);
+        if (mapping.isPresent()) {
+            switch (legacyGenericColumnMapping) {
+                case ENABLE:
+                    return mapping;
+                case THROW:
```

Comment:
Can you please provide some tests for this and for ignore?

Reply:
No connector should allow test covering that, so i would need to have a whole new connector just for this test.